pkgs/native_toolchain_c/lib/src/cbuilder/clinker.dart

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:logging/logging.dart';
+import 'package:meta/meta.dart';
+import 'package:native_assets_cli/native_assets_cli.dart';
+
+import 'ctool.dart';
+import 'language.dart';
+import 'linker_options.dart';
+import 'linkmode.dart';
+import 'output_type.dart';
+import 'run_cbuilder.dart';
+
+export 'linker_options.dart';
+
+/// Specification for linking an artifact with a C linker.
+class CLinker extends CTool implements Linker {
+  final LinkerOptions linkerOptions;
+
+  CLinker.library({
+    required super.name,
+    super.assetName,
+    required this.linkerOptions,
+    super.sources = const [],
+    super.includes = const [],
+    super.frameworks = CTool.defaultFrameworks,
+    @visibleForTesting super.installName,
+    super.flags = const [],
+    super.defines = const {},
+    super.pic = true,
+    super.std,
+    super.language = Language.c,
+    super.cppLinkStdLib,
+    super.linkModePreference,
+  }) : super(type: OutputType.library);
+
+  /// Runs the C Linker with on this C build spec.
+  ///
+  /// Completes with an error if the linking fails.
+  @override
+  Future<void> run({
+    required LinkConfig config,
+    required LinkOutput output,
+    required Logger? logger,
+  }) async {
```

Comment:
If this only supports Linux, please check the targetOS and throw an `UnsupportedError` for the other target OSes.

Also, add tests that expect an unsupported error for the other OSes.